File: Basic_maths/C++/print_all_divisors.cpp
#include<iostream>
#include<math.h>
#include<vector>
#include<algorithm>

using namespace std;


void divisor(int n){
    
    vector<int> list;

    for(int i=1;i<=sqrt(n);i++){

        if( n % i == 0){
            list.push_back(i);
            // cout<<"Divisor : " << i << endl;
            if(i!=n/i){
                // cout<<"Divisor : " << n/i << endl;
                list.push_back(n/i);
            }
        }
    }

    sort(list.begin(),list.end());

    for(auto l:list){
        
        cout<<"Divisors : " << l  << endl;
    }

    
}

int main(){

    int n=36;
    // int c=0;
    // for(int i=1;i<=36;i++){

    //     if( n % i == 0){

    //         cout<<"Divisor : " << i << endl;
    //         c++;
    //     }
    // }

    // cout<<c;
    

    divisor(n);
    return 0;
}

File: Basic_maths/Python/print_all_divisors.py

def no_divisor(n):

    c=0
    for i in range(1,n+1):
        if n % i ==0:
            c+=1
    return c

import math
def no_divisor_optimal(n):

    c=0
    for i in range(1,int(math.sqrt(n))):
        if n % i ==0:
            c+=1
        
        if i!=n//i:
            c+=1
    return c

def divisors_optimal(n):

    c=0
    divisors=[]
    for i in range(1,int(math.sqrt(n))):
        if n % i ==0:
            divisors.append(i)
        
        if i!=n//i:
            divisors.append(n//i)

    return sorted(divisors)



n=36

print("Number of Divisor : ",no_divisor(n))
print("Number of Divisor : ",no_divisor_optimal(n))
print("Number of Divisor : ",divisors_optimal(n))